cmd/frontend/graphqlbackend/repository_comparison.go

```diff
@@ -104,37 +104,31 @@ func NewRepositoryComparison(ctx context.Context, r *RepositoryResolver, args *R
 		return nil, err
 	}
 
-	var (
-		wg               sync.WaitGroup
-		base, head       *GitCommitResolver
-		baseErr, headErr error
-	)
+	head, err := getCommit(ctx, *grepo, headRevspec)
+	if err != nil {
+		return nil, err
+	}
 
-	wg.Add(2)
-	go func() {
-		defer wg.Done()
-		// Find the common merge-base for the diff. That's the revision the diff applies to,
-		// not the baseRevspec.
-		mergeBaseCommit, err := git.MergeBase(ctx, *grepo, api.CommitID(baseRevspec), api.CommitID(headRevspec))
-		if err != nil {
-			baseErr = err
-			return
-		}
-		// We use the merge-base as the base commit here, as the diff will only be guaranteed to be
-		// applicable to the file from that revision.
-		commitString := strings.TrimSpace(string(mergeBaseCommit))
-		base, baseErr = getCommit(ctx, *grepo, commitString)
-	}()
-	go func() {
-		defer wg.Done()
-		head, headErr = getCommit(ctx, *grepo, headRevspec)
-	}()
-	wg.Wait()
-	if baseErr != nil {
-		return nil, baseErr
-	}
-	if headErr != nil {
-		return nil, headErr
+	// Make sure we call ResolveRevision before we find the common merge-base, in case
+	// the commits don't exist locally.
+	_, err = git.ResolveRevision(ctx, *grepo, nil, baseRevspec, git.ResolveRevisionOptions{NoEnsureRevision: false})
```

Comment:
The default value of a boolean in a struct in Go is `false` so there's no need to initialise it. Exception: often helps readability to make a value explicitly false. But since we _do_ want to ensure that the revision exists locally, it's more confusing to read the double-negative than not.

```suggestion
	_, err = git.ResolveRevision(ctx, *grepo, nil, baseRevspec, git.ResolveRevisionOptions{})
```